### PR TITLE
CHE-7836: Update Che JDT-LS workspace on project creation/deletion

### PR DESCRIPTION
**Pull request by tolusha**
### What does this PR do?
Update Che JDT-LS workspace on project creation/deletion

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/7836

### CODE REVIEW

File: plugins/plugin-java/che-plugin-java-ext-lang-shared/src/main/java/org/eclipse/che/ide/ext/java/shared/Constants.java [hunk 1]
```diff
@@ -18,7 +18,7 @@
   // project categories
   public static final String JAVA_CATEGORY = "Java";
   public static final String JAVA_ID = "java";
-  // project attribute names
+  // project attribute names2
```
Comment (svor) [inline plugins/plugin-java/che-plugin-java-ext-lang-shared/src/main/java/org/eclipse/che/ide/ext/java/shared/Constants.java]: why is it names**2**?

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java [hunk 2]
```diff
@@ -477,46 +508,42 @@ private void fixLocation(ExtendedSymbolInformation symbol) {
...
-              new com.google.common.reflect.TypeToken<
-                  List<Either<String, ResourceLocation>>>() {}.getType());
-      Either<String, ResourceLocation> l = location.get(0);
-
-      if (l.isLeft()) {
-        return new LocationImpl(l.getLeft(), lineNumber, null);
-      } else {
-        return new LocationImpl(
-            l.getRight().getFqn(), lineNumber, true, l.getRight().getLibId(), null);
+  public JobResult updateWorkspace(UpdateWorkspaceParameters updateWorkspaceParameters) {
```
Comment (svor) [inline plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java]: pls add javadoc

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java [hunk 3]
```diff
@@ -477,46 +508,42 @@ private void fixLocation(ExtendedSymbolInformation symbol) {
...
+        jobResult.setMessage(
+            "Skipped. Language server not initialized. Workspace updating is not required.");
+        return jobResult;
       }
-    } catch (JsonSyntaxException | InterruptedException | ExecutionException | TimeoutException e) {
-      throw new JsonRpcException(-27000, e.getMessage());
     }
+
+    Type type = new TypeToken<JobResult>() {}.getType();
+    return doGetOne(Commands.UPDATE_WORKSPACE, singletonList(updateWorkspaceParameters), type, 60);
```
Comment (svor) [inline plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java]: We have some constant about timeout here `org.eclipse.che.ide.ext.java.shared.Constants`, maybe move this number or change existed. What do you think?

File: plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/ProjectsListener.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import org.eclipse.che.api.project.server.notification.ProjectInitializedEvent;
+import org.eclipse.che.commons.lang.concurrent.LoggingUncaughtExceptionHandler;
+import org.eclipse.che.jdt.ls.extension.api.dto.JobResult;
+import org.eclipse.che.jdt.ls.extension.api.dto.UpdateWorkspaceParameters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** @author Anatolii Bazko */
+@Singleton
+public class ProjectsListener {
```
Comment (svor) [inline plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/ProjectsListener.java]: pls add javadoc about in which case it is useful

**Review by svor — APPROVED** [review]

**Review by tsmaeder — APPROVED** [review]

**Review by mmorhun — APPROVED** [review]

**Pull request merged**